src/main/java/eu/dissco/orchestration/backend/web/FdoRecordBuilder.java

```diff
@@ -0,0 +1,86 @@
+package eu.dissco.orchestration.backend.web;
+
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.DIGITAL_OBJECT_TYPE;
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.FDO_PROFILE;
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.ISSUED_FOR_AGENT;
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.MAS_NAME;
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.SOURCE_DATA_STANDARD;
+import static eu.dissco.orchestration.backend.domain.FdoProfileAttributes.SOURCE_SYSTEM_NAME;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import eu.dissco.orchestration.backend.domain.MachineAnnotationService;
+import eu.dissco.orchestration.backend.domain.Mapping;
+import eu.dissco.orchestration.backend.domain.ObjectType;
+import eu.dissco.orchestration.backend.domain.SourceSystem;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class FdoRecordBuilder {
```

Comment:
As we are not using the Builder pattern it would be better to call this something else, such as FdoRecordComponent or FdoRecordService